app/models/concerns/searchable.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Searchable
+  extend ActiveSupport::Concern
+
+  included do
+    SEARCHABLE_FIELDS = %w(title description author)
+
+    SEARCHABLE_FIELDS.each do |attr|
+      scope "search_by_#{attr}".to_sym, ->(value) do
+        where("#{attr} LIKE ?", "%#{value}%")
+      end
+    end
+  end
+
+  module ClassMethods
+    def search(keyword:)
+      outcome = []
+
+      SEARCHABLE_FIELDS.each do |key|
+        results = self.public_send("search_by_#{key}", keyword).as_json
+        next if results.blank?
+
+        results.each { |result| outcome << result }
```

Comment:
Another issue with this approach is that this creates 3 different PSQL transactions. Some suggestions:

1. Simplify the `searchable` logic. This should produce only `1 SQL transaction`

```
fields_query = SEARCHABLE_FIELDS.map { |f| "#{f} ilike :query" }
self.where("#{fields_query.join(" OR ")}", query: keyword)
```

2. Make use of gems that does full-text search
https://github.com/Casecommons/pg_search